src/quirks/mod.rs

```diff
@@ -0,0 +1,104 @@
+use http::{header, Method};
+use regex::Regex;
+use reqwest::{Request, Url};
+
+/// Sadly some pages only return plaintext results if Google is trying to crawl them.
+const GOOGLEBOT: &str = "Mozilla/5.0 (compatible; Googlebot/2.1; +http://google.com/bot.html)";
+
+#[derive(Debug, Clone)]
+pub struct Quirk {
+    pub pattern: Regex,
+    pub rewrite: fn(Request) -> Request,
+}
+
+#[derive(Debug, Clone)]
+pub struct Quirks {
+    quirks: Vec<Quirk>,
+}
+
+impl Quirks {
+    pub fn init() -> Self {
```

Comment:
Why not `new`?

If the intention was to indicate that it's going to "init standard quirks" then maybe we should be a bit more verbose with the name, e.g.

`pub fn standard_quirks() -> Self {`

It's just a nit-picking suggestion though.

Reply:
You're right. Come to think of it, there is a default way to handle that scenario and that is... 🥁 🥁 🥁 the `Default` trait. 🎉 
So I'll implement that instead. `Quirks::init()` becomes `Quirk::default()` then, which is closer to what I'd like to convey.